src/main/java/com/avispl/symphony/dal/communicator/middleatlantic/SocketCommunicator.java

```diff
@@ -0,0 +1,460 @@
+package com.avispl.symphony.dal.communicator.middleatlantic;
+
+import com.avispl.symphony.api.dal.dto.control.ConnectionState;
+import com.avispl.symphony.api.dal.error.CommandFailureException;
+import com.avispl.symphony.dal.BaseDevice;
+import com.avispl.symphony.dal.communicator.Communicator;
+import com.avispl.symphony.dal.communicator.ConnectionStatus;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * TCP Socket communicator
+ * @author Jonathan.Gillot
+ *
+ * */
+public class SocketCommunicator extends BaseDevice implements Communicator {
```

Comment:
Didn't we have a task to move SocketCommunicator to DAL commons instead of copying it into each adapter? What happened to that task? It is OK to copy for now, just want to make sure that task is not lost.

Reply:
Yes, that issue is in the backlog and we were planning to add it into the next sprint (We really need this to get into the next release, so there's no way we're postponing this.). 
https://avi-spl.atlassian.net/browse/SYM-176